File: src/types/enums.ts
/**
 * Supported content types for a source document.
 * 
 * Each content type is handled by a specific extractor in src/services/ingestion/extractors:
 * - Text: TextExtractor - Plain text content
 * - Url: UrlExtractor - Web content from URLs  
 * - Pdf: PdfExtractor - PDF document text extraction
 * - Image: ImageExtractor - OCR and image analysis (planned)
 * - Video: VideoExtractor - Video transcription (planned)
 * - Audio: AudioExtractor - Audio transcription (planned)
 * - Conversation: ConversationExtractor - Chat/conversation formats
 */
export enum ContentType {
  Text = "text",
  Url = "url",
  Pdf = "pdf",
  Image = "image",
  Video = "video",
  Audio = "audio",
  Conversation = "conversation"
}

/** Memory categorization used by Memento v2. */
export enum MemoryType {
  Fact = "fact",
  Preference = "preference",
  Episode = "episode",
  Derived = "derived"
}

/** Processing lifecycle for ingested documents. */
export enum DocumentStatus {
  Queued = "queued",
  Extracting = "extracting",
  Chunking = "chunking",
  ExtractingMemories = "extracting_memories",
  Embedding = "embedding",
  Indexing = "indexing",
  Done = "done",
  Error = "error"
}

/** Intelligent relation types between memories. */
export enum RelationType {
  Updates = "UPDATES",
  Extends = "EXTENDS",
  Derives = "DERIVES",
  ExtractedFrom = "EXTRACTED_FROM"
}

File: src/types/enums.py
from enum import Enum


class ContentType(str, Enum):
    """Supported content types for a source document."""

    TEXT = "text"
    URL = "url"
    PDF = "pdf"
    IMAGE = "image"
    VIDEO = "video"
    AUDIO = "audio"
    CONVERSATION = "conversation"


class MemoryType(str, Enum):
    """Memory categorization used by Memento v2."""

    FACT = "fact"
    PREFERENCE = "preference"
    EPISODE = "episode"
    DERIVED = "derived"


class DocumentStatus(str, Enum):
    """Processing lifecycle for ingested documents."""

    QUEUED = "queued"
    EXTRACTING = "extracting"
    CHUNKING = "chunking"
    EXTRACTING_MEMORIES = "extracting_memories"
    EMBEDDING = "embedding"
    INDEXING = "indexing"
    DONE = "done"
    ERROR = "error"


class RelationType(str, Enum):
    """Intelligent relation types between memories."""

    UPDATES = "UPDATES"
    EXTENDS = "EXTENDS"
    DERIVES = "DERIVES"
    EXTRACTED_FROM = "EXTRACTED_FROM"
